src/Algebra/Graph/Class.hs

```diff
@@ -155,7 +155,7 @@ instance Dioid e => Graph (LG.Graph e a) where
     overlay = LG.overlay
     connect = LG.connect one
 
-instance (Dioid e, Eq e, Ord a) => Graph (LAM.AdjacencyMap e a) where
+instance (Dioid e, Ord a) => Graph (LAM.AdjacencyMap e a) where
```

Comment:
This is a little frustrating but probably acceptable.  Technically, `LAM.AdjacencyMap` seems like it could be a `Graph` so long as it has the multiplicative monoid even if it doesn't have the additive one (that is, it has `<.>` and `one` but doesn't have `zero`).  Thus, the `Dioid e` constraint is too strong.  However, there's no nice way to represent this, so it seems like we should just require the full `Dioid e` constraint.